plugins/inara.py

```diff
@@ -757,30 +954,42 @@ def journal_entry(cmdr, is_beta, system, station, entry, state):
         this.system_link.update_idletasks()
 
     if config.get('station_provider') == 'Inara':
-        this.station_link['text'] = this.station or (this.system_population and this.system_population > 0 and STATION_UNDOCKED or '')
+        to_set = this.station
+        if not to_set and this.system_population is not None:
+            if this.system_population > 0:
+                to_set = STATION_UNDOCKED
+            else:
+                to_set = ''
+
+        this.station_link['text'] = to_set
         this.station_link['url'] = station_url(this.system, this.station)
         this.station_link.update_idletasks()
 
+
 def cmdr_data(data, is_beta):
     this.cmdr = data['commander']['name']
 
     # Always store initially, even if we're not the *current* system provider.
     if not this.station_marketid:
         this.station_marketid = data['commander']['docked'] and data['lastStarport']['id']
+
     # Only trust CAPI if these aren't yet set
-    this.system = this.system or data['lastSystem']['name']
-    this.station = this.station or data['commander']['docked'] and data['lastStarport']['name']
+    this.system = this.system if this.system else data['lastSystem']['name']
+    this.station = data['lastStarport']['name'] if data['commander']['docked'] and not this.station else this.station
```

Comment:
`this.station = this.station if this.station elif data['commander']['docked'] data['lastStarport']['name']`
Or is elif not allowed on such embedded conditionals ?

Basically it should be more obvious that `this.station` will stay as that if it's set.

Heck, just make it multi-line:

```
if not this.station and data['commander']['docked']:
    this.station = data['lastStarport']['name']
```
If this.station isn't None then we leave it be.

Reply:
elif is not allowed in conditionals like that no -- will do as suggested